compiler/rustc_symbol_mangling/src/typeid.rs

```diff
@@ -0,0 +1,25 @@
+// For more information about type metadata and type metadata identifiers for cross-language LLVM
+// CFI support, see the design document at: https://rcvalle.com/docs/rust-cfi-design-doc.pdf.
+
+use rustc_middle::ty::{FnSig, Ty, TyCtxt};
+use rustc_target::abi::call::FnAbi;
+
+mod typeid_itanium_cxx_abi;
+
+// Returns a type metadata identifier for the specified FnAbi.
+pub fn typeid_for_fnabi<'tcx>(tcx: TyCtxt<'tcx>, fn_abi: &FnAbi<'tcx, Ty<'tcx>>) -> Option<String> {
+    if tcx.sess.is_sanitizer_cfi_enabled() {
```

Comment:
I suspect the responsibility for checking if the `CFI` is enabled should be for the caller of this function. It doesn't sound to me like typeids and such are super specific to CFI in particular.

Reply:
Makes sense. Done.